projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts

```diff
@@ -142,10 +148,16 @@ export class IgxItemListDirective {
 
         const listName = (event.target as HTMLElement).className;
 
-        if (listName.indexOf('minuteList') !== -1) {
+        if (listName.indexOf('minuteList') !== -1 && this.timePicker.hourList) {
             this.timePicker.hourList.nativeElement.focus();
-        } else if (listName.indexOf('ampmList') !== -1) {
-            this.timePicker.minuteList.nativeElement.focus();
+        }
+
+        if (listName.indexOf('ampmList') !== -1) {
+            if (this.timePicker.minuteList) {
+                this.timePicker.minuteList.nativeElement.focus();
+            } else if (this.timePicker.hourList) {
+                this.timePicker.hourList.nativeElement.focus();
```

Comment:
duplicate code with the line above (...hourList.nativeElement.focus()) - merge if possible